pkg/execution/executor/service.go

```diff
@@ -168,17 +168,21 @@ func (s *svc) getFinishHandler(ctx context.Context) (func(context.Context, sv2.I
 
 func (s *svc) Run(ctx context.Context) error {
 	logger.From(ctx).Info().Msg("subscribing to function queue")
-	return s.queue.Run(ctx, func(ctx context.Context, info queue.RunInfo, item queue.Item) error {
+	return s.queue.Run(ctx, func(ctx context.Context, info queue.RunInfo, item queue.Item) (bool, error) {
```

Comment:
[minor] I would prefer to introduce some sort of `result` type; in a lot of queue code we return sets of bools, ints, errors, etc.. This (a) requires care on the caller's part and (b) allows room to return invalid states (what does it mean if we return (`true, error(…)` from this function?).

Introducing more structured results ([`Either3`](https://github.com/samber/mo/blob/master/either3.go#L44) for inspiration) would reduce the room for programming errors and eliminate the need to handle invalid states.